ethcore/src/engines/tendermint/mod.rs

```diff
@@ -101,6 +102,60 @@ pub struct Tendermint {
 	validators: Box<ValidatorSet>,
 }
 
+struct EpochVerifier {
+	subchain_validators: SimpleList,
+}
+
+impl super::EpochVerifier for EpochVerifier {
+	fn verify_light(&self, header: &Header) -> Result<(), Error> {
+		let message = header.bare_hash();
+
+		let mut addresses = HashSet::new();
+		let ref header_signatures_field = header.seal().get(2).ok_or(BlockError::InvalidSeal)?;
+		for rlp in UntrustedRlp::new(header_signatures_field).iter() {
+			let signature: H520 = rlp.as_val()?;
+			let address = public_to_address(&recover(&signature.into(), &message)?);
+
+			if !self.subchain_validators.contains(header.parent_hash(), &address) {
+				return Err(EngineError::NotAuthorized(address.to_owned()).into());
+			}
+			addresses.insert(address);
+		}
+
+		let n = addresses.len();
+		let threshold = self.subchain_validators.len() * 2/3;
+		if n > threshold {
```

Comment:
Does it matter if it's ">" or ">="? I'd imagine in the case of three `subchain_validators` we'd want verification to pass when we get 2 of them